### DISCUSSION TITLE
Value of Using Graphite

### DISCUSSION THREAD

**Discussion by technoplato** [ephemeraHQ/converse-app#1013] [Show and tell]
I've advocated for the usage of Graphite on our Mobile team so below I provide some information about why I feel that it is a useful tool. 

Businessy clickbait pitch: https://www.youtube.com/watch?v=6xpwYgUffmM

20 minute product tour: https://www.youtube.com/watch?v=sBcd9uopLOY

Value of Stacking PRs: https://www.stacking.dev

**TLDW**: Stacking PRs allows you to build complex features with smaller PRs. Graphite is a tool that helps you build stacked PRs and merge upstream changes when requested without the hassle of rebasing manually (which is often error prone, especially when you have to keep up with `develop` branch.

**Comment by thierryskoda** [comment]
ok watched the videos. Seems cool but I don't think we're ready to introduce this IMO. I feel there are enough things to learn and fix in our codebase that introducing another thing we have to learn and use correctly will make my head explode lol.

**Comment by neekolas** [comment]
I don't think it's a decision that has to be made at the team level. 

In xmtpd, Rich and I use Graphite. Martin doesn't. That's perfectly fine. Use the tools that are right for you. We have a team account, so you don't need approval to give it a shot. 

The one exception to this is the merge queue feature. Having a repo where some PRs use it and some don't can compound merge conflicts and make them harder to resolve. I think I'm going to turn that feature off for the two repos we experimented with it. We don't have quite enough PRs to see the true benefits anyways.
